java_socketio_chatroom_client/src/test/java/io/github/marcodiri/java_socketio_chatroom_client/ChatroomClientTest.java

```diff
@@ -0,0 +1,124 @@
+package io.github.marcodiri.java_socketio_chatroom_client;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.assertj.core.api.Assertions.fail;
+import static org.awaitility.Awaitility.await;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+import java.net.URI;
+import java.sql.Timestamp;
+
+import org.json.JSONObject;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import io.github.marcodiri.java_socketio_chatroom_client.model.Message;
+import io.github.marcodiri.java_socketio_chatroom_client.view.ClientView;
+import io.github.marcodiri.java_socketio_chatroom_server_mock.ChatroomServerMock;
+import io.socket.client.IO;
+
+public class ChatroomClientTest {
+
+	private ClientView view;
+
+	private ChatroomClient client;
+
+	private ChatroomServerMock serverMock;
+
+	@Before
+	public void setup() {
+		view = mock(ClientView.class);
```

Comment:
Had to do this instead of @Mock because the constructor does not require only mocks.

Reply:
I agree